### PR TITLE
wolfssl: Add version 5.6.0.

### PR DESCRIPTION
**Pull request by 0xFireWolf**
Specify library name and version:  **wolfssl/5.6.0**

This PR contains the following changes:
- Add the latest version 5.6.0.
- Link against the frameworks, `CoreFoundation` and `Security` when building the static library for Apple platforms, fixing the linker errors related to undefined symbols.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/wolfssl/all/conanfile.py [hunk 1]
```diff
@@ -156,4 +156,6 @@ def package_info(self):
             if self.settings.os in ["Linux", "FreeBSD"]:
                 self.cpp_info.system_libs.extend(["m", "pthread"])
             elif self.settings.os == "Windows":
-                self.cpp_info.system_libs.extend(["advapi32", "ws2_32"])
+                self.cpp_info.system_libs.extend(["advapi32", "ws2_32", "crypt32"])
```
Comment (RubenRBS) [inline recipes/wolfssl/all/conanfile.py:159]: Is this line a bugfix? Was it needed for previous versions?

**Review by RubenRBS — COMMENTED** [review]
Hi! Thanks a lot for your contribution. I only have one minor question before approving :)

File: recipes/wolfssl/all/conanfile.py (hunk 1 above)
Comment (0xFireWolf) [reply recipes/wolfssl/all/conanfile.py]: Thanks for reviewing this PR. I just double checked and found that `crypt32` was not needed in previous versions. Also, `CoreFoundation` and `Security` frameworks are not needed in previous versions.

**Review by ericLemanissier — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]
Dont mind me just trying to test some code 🧑‍💻

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
GitHub no longer dismissed the community reviewers 🤔

**Review by prince-chrismc — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Pull request merged**